hwdef: Add JFB200 of JAE flight controllers (#32786)
ArduPilot/ardupilot: libraries/AP_HAL_ChibiOS/hwdef/JFB200/README.md

Some of the sensors mounted on the octa module are vibration-isolated and feature built-in heaters.  
![JFB-200 overview](JFB-200_overview.png)

## Features

- Processor

  - STM32H755 microcontroller

- Sensors

  - Three IMUs: ICM45686, ASM330 and IIM42653 SPI IMU
  - Two BAROs: two BMP390 SPI barometer
  - Two Mags: builtin I2C BMM350 and IST8310 magnetometer

- Interfaces

  - microSD card slot
  - 7 UARTs plus USB (including GPS1, GPS2, RCIN and S.OUT)
  - 16 PWM outputs (8 PWM shares GPIO)
  - Three I2C and two CAN ports
  - Two external Buzzer (Open/Drain and 24V Out)
  - external safety Switch
  - voltage monitoring for servo rail and Vcc
  - Ethernet port

- Power

  - two dedicated power input ports for external power bricks
  - two analog battery voltage and current sensing port

## Pinout

![JFB-200 pinout](JFB-200_pinout.png)

## UART Mapping

- SERIAL0 -> USB    (MAVLink2)
- SERIAL1 -> UART7  (TELEM1, MAVLink2, DMA-enabled)
- SERIAL2 -> UART5  (TELEM2, MAVLink2, DMA-enabled)
- SERIAL3 -> USART3 (TELEM3, MAVLink2, DMA-enabled)
- SERIAL4 -> USART1 (GPS1, DMA-enabled)
- SERIAL5 -> USART2 (GPS2, DMA-enabled)
- SERIAL6 -> UART8  (RCIN, DMA-enabled)

> - SERIAL7 -> USART6 (S.OUT, DMA-enabled)

Reviewer: ```suggestion
- SERIAL7 -> USART6 (SBUSOUT, DMA-enabled)
```

Author: Hi @Hwurzburg ,

Thanks,

SBUS is a registered trademark of Futaba Corporation.
To avoid unauthorized use, we have changed SBUSOUT to SOUT.

Reviewer: A trademark search shows that "SBUS" or "SBUSOUT" is not a registered trademark in the USPTO, WIPO, EUIPO. or JPO databases....SBUS is a freely used term in RC documentation..Futaba does trademark other terms like  FASST, etc.

Author: Hi @Hwurzburg ,

Thanks for about trademark of SBUSOUT.

I see this. We have changed to SBUSOUT.

- SERIAL8 -> USB    (MAVLink2)

The TELEM1, TELEM2 and TELEM3 ports have RTS/CTS pins, the other UARTs do not have RTS/CTS.

## RC Input

RC input is configured on the port marked RCIN/UART.  

> This connector supports all RC protocols. Two cables are available for this port.  

Reviewer: ```suggestion
This connector supports all RC unidirectional RC protocols. Two cables are available for this port.  
```

To use software binding of Spektrum satellite receivers you need to use the Spektrum satellite cable.

> 

Reviewer: ```suggestion
To used bi-dectional protocols such as CRSF/ELRS, a full UART must be used instead. UART8 is pre-configure for these RC inputs. See [RC systems](https://ardupilot.org/copter/docs/common-rc-systems.html) for more details.
```

## PWM Output

> The JFB-200 supports up to 16 PWM outputs.

Reviewer: ```suggestion
The JFB-200 supports up to 16 PWM/DShot outputs.
```

The 16 PWM outputs are in 4 groups:  

- PWM  1,  2,  3 and  4 in group1 (TIM1)
- PWM  5,  6,  7 and  8 in group2 (TIM3)
- PWM  9, 10, 11 and 12 in group3 (TIM4)
- PWM 13, 14, 15 and 16 in group4 (TIM8)

Channels within the same group need to use the same output rate.  
If any channel in a group uses DShot then all channels in the group need to use DShot.  
PWM output voltage can be changed setting BRD_PWM_VOLT_SEL parameter.  

## Battery Monitoring

The board has two dedicated power monitor ports on 8 pin connectors.  
The correct battery setting parameters are dependent on the type of power brick which is connected.  

> Recomended input voltage is 4.9 to 5.5 volt.

Reviewer: you set defaults in hwdef for both monitors....list them here...special not about scales being set to "1" which need to be set by user (usually better to allow the defaults to be used instead of "1")

Author: I modified hwdef.

Reviewer: hwdef still has set and enabled two analog monitors with some defaults as "1"...my original comment still applies

Author: Hi, @Hwurzburg.

I apologize for not understanding the meaning of your comment.
I added a list of default parameters for the battery monitor.

## Compass

The JFB-200 has two builtin compasses BMM350 and IST8310.  
Due to potential interference the board is usually used with an external I2C compass as part of a GPS/Compass combination.

## GPIOs

> The 8 PWM ports (PMW 9-16) can be used as GPIOs (relays, buttons, RPM etc).  

Reviewer: ```suggestion
8 PWM ports (PMW 9-16) can be used as GPIOs (relays, buttons, RPM etc).  PWM 1-8 can be used as GPIO outputs only for realys,etc.
```

PWM1-8 are connected via dedicated output buffers. PWM 9-16 are connected via bidirectional buffers.  

> Therefore, only PWM 9-16 GPIO pins are supported.

Reviewer: remove...incorrect

Author: I removed this.

The numbering of the GPIOs for PIN variables in ArduPilot is:  

- PWM(9)  50
- PWM(10) 51
- PWM(11) 52
- PWM(12) 53
- PWM(13) 54
- PWM(14) 55
- PWM(15) 56
- PWM(16) 57
- CAP1 58
- CAP2 59

## Analog inputs

The JFB-200 has 10 analog inputs

> - ADC Pin2  -> +3.3V Sens

Reviewer: ```suggestion
- ADC Pin2  -> +3.3V Sense
```

- ADC Pin4  -> Battery Voltage 2
- ADC Pin6  -> Battery Current Sensor 2
- ADC Pin8  -> ADC IN 1 (6.6V)
- ADC Pin10 -> RSSI voltage monitoring
- ADC Pin13 -> ADC IN 2 (3.3V)
- ADC Pin14 -> SERVORAIL sens

> - ADC Pin15 -> 5V sens

Reviewer: ```suggestion
- ADC Pin15 -> 5V sense
```

- ADC Pin16 -> Battery Voltage
- ADC Pin18 -> Battery Current Sensor

## I2C Buses

The JFB-200 has 4 I2C interfaces.
I2C 3 is for internal only.

- the internal I2C port  is bus 3 in ArduPilot (I2C3 in hardware)
- the port labelled GPS1 is bus 1 in ArduPilot (I2C1 in hardware)
- the port labelled GPS2 is bus 2 in ArduPilot (I2C2 in hardware)
- the port labelled I2C  is bus 4 in ArduPilot (I2C4 in hardware)

## CAN

The JFB-200 has two independent CAN buses with terminating resistors.

## Dedicated Signal input/output

> The JFB-200 has the following dedicated discrete signals

Reviewer: ```suggestion
The JFB-200 has the following dedicated discrete signals on the AUX port
```

| --- | --- | --- | --- |
| 1 | VCC | +5V |
| 2 | UART5_TX (OUT) | +3.3V |
| 3 | UART5_RX (IN) | +3.3V |
| 4 | UART5_CTS | +3.3V |
| 5 | UART5_RTS | +3.3V |
| 6 | GND ||

### TELEM3 ports

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC | +5V |
| 2 | USART3_TX (OUT) | +3.3V |
| 3 | USART3_RX (IN) | +3.3V |
| 4 | USART3_CTS | +3.3V |
| 5 | USART3_RTS | +3.3V |
| 6 | GND ||

### GPS2 port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC | +5V |
| 2 | USART2_TX (OUT) | +3.3V |
| 3 | USART2_RX (IN) | +3.3V |
| 4 | I2C2 SCL | +3.3V |
| 5 | I2C2 SDA | +3.3V |
| 6 | GND | |

### ETH port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | RXN | |
| 2 | RXP | |
| 3 | TXN | |
| 4 | TXP | |

### GPS1 port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC | +5V |
| 2 | USART1_TX (OUT) | +3.3V |
| 3 | USART1_RX (IN) | +3.3V |
| 4 | I2C1 SCL | +3.3V(pullup) |
| 5 | I2C1 SDA| +3.3V(pullup) |
| 6 | Safety Button | +3.3V |
| 7 | Safety LED | +3.3V |
| 8 | VCC3 | +3.3V |
| 9 | BUZZER | OPEN/DRAIN |
| 10 | GND |  |

### BUZZER port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |

> | 1 | BUZZER VCC | +24V/GND |

Reviewer: 24V???? really?where does this come from?

Author: Yes, it's true.
This connector is for a buzzer that supports 24V and outputs 24V.

Reviewer: so it will destroy a 5V buzzer if connected...readme needs a warning added!

Author: I added the note.

| 2 | GND | |

### RCIN/UART port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCChigh | +5V |
| 2 | USART6_TX (OUT) | +3.3V | S.OUT |
| 3 | USART6_RX (IN) | +3.3V |
| 4 | RSSI | +3.3V |
| 5 | PPM | +3.3V | Low Reso |

> | 6 | USART8_TX (OUT) | +3.3V |

Reviewer: ```suggestion
| 6 | USART8_TX (OUT) | +3.3V | RCIN |
```

| 7 | USART8_RX (IN) | +3.3V | RCIN |
| 8 | GND | |

### I2C port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC | +5V |
| 2 | I2C4 SCL | +3.3V |
| 3 | I2C4 SDA | +3.3V |
| 4 | GND | |

### USB port

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC | +5V | Power IN |
| 2 | D_Minus |  |
| 3 | D_Plus | |
| 4 | GND | |

### RCIN port

2.54mm pitch pin header

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | RCIN | +3.3V |
| 2 | VCChigh | +5V |
| 3 | GND | |

> ### S.OUT port

Reviewer: ```suggestion
### SOUT port
```
to match marking

2.54mm pitch pin header

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | S.OUT | +3.3V |
| 2 | VCChigh | +5V |
| 3 | GND | |

### POWER1, POWER2 ports

Molex picoblade 5024430670

| Pin | Signal | Volt | Remarks |
| --- | --- | --- | --- |
| 1 | VCC IN | +4.9V ～ +5V | Power IN |
| 2 | VCC IN | +4.9V ～ +5V | Power IN |
| 3 | CURRENT | +3.3V |
| 4 | VOLTAGE | +3.3V |
| 5 | GND | |
| 6 | GND | |

> 

Reviewer: add a Firmware header and state where firmware is located...see other readmes

Author: I add  a Firmware header.

Reviewer: you added a comment under the loading firmware header...this is acceptable also...

  - 16 PWM outputs (16 PWM shares GPIO)
  - Three I2C and two CAN ports
  - Two external Buzzer (Open/Drain and 24V Out)
  - external safety Switch
  - voltage monitoring for servo rail and Vcc
  - Ethernet port

- Power

  - two dedicated power input ports for external power bricks
  - two analog battery voltage and current sensing port

## Pinout

![JFB-200 pinout](JFB-200_pinout.png)

## UART Mapping

- SERIAL0 -> USB    (MAVLink2)
- SERIAL1 -> UART7  (TELEM1, MAVLink2, DMA-enabled)
- SERIAL2 -> UART5  (TELEM2, MAVLink2, DMA-enabled)
- SERIAL3 -> USART3 (TELEM3, MAVLink2, DMA-enabled)
- SERIAL4 -> USART1 (GPS1, DMA-enabled)
- SERIAL5 -> USART2 (GPS2, DMA-enabled)
- SERIAL6 -> UART8  (RCIN, DMA-enabled)
- SERIAL7 -> USART6 (SOUT, DMA-enabled)
- SERIAL8 -> USB    (MAVLink2)

The TELEM1, TELEM2 and TELEM3 ports have RTS/CTS pins, the other UARTs do not have RTS/CTS.

## RC Input

RC input is configured on the port marked RCIN/UART.  
This connector supports all RC unidirectional RC protocols. Two cables are available for this port.  
To use software binding of Spektrum satellite receivers you need to use the Spektrum satellite cable.

> To used bi-dectional protocols such as CRSF/ELRS, a full UART must be used instead. UART8 is pre-configure for these RC inputs. See [RC systems](https://ardupilot.org/copter/docs/common-rc-systems.html) for more details.

Reviewer: sorry my typos...

```suggestion
To use bi-directional protocols such as CRSF/ELRS, a full UART must be used instead. UART8 is pre-configured for these RC inputs. See [RC systems](https://ardupilot.org/copter/docs/common-rc-systems.html) for more details.
```

## PWM Output

The JFB-200 supports up to 16 PWM/DShot outputs.

The 16 PWM outputs are in 4 groups:  

- PWM  1,  2,  3 and  4 in group1 (TIM1)
- PWM  5,  6,  7 and  8 in group2 (TIM3)
- PWM  9, 10, 11 and 12 in group3 (TIM4)
- PWM 13, 14, 15 and 16 in group4 (TIM8)

Channels within the same group need to use the same output rate.  
If any channel in a group uses DShot then all channels in the group need to use DShot.  
PWM output voltage can be changed setting BRD_PWM_VOLT_SEL parameter.  

## Battery Monitoring

The board has two dedicated power monitor ports on 8 pin connectors.  
The correct battery setting parameters are dependent on the type of power brick which is connected.  
Recomended input voltage is 4.9 to 5.5 volt.

## Compass

The JFB-200 has two builtin compasses BMM350 and IST8310.  
Due to potential interference the board is usually used with an external I2C compass as part of a GPS/Compass combination.

## GPIOs

> PWM 1-8 can be used as GPIO outputs only for realys,etc.  

Reviewer: ```suggestion
PWM 1-8 can be used as GPIO outputs only for relays,etc.  
```

PMW 9-16 can be used as GPIOs for relays, buttons, RPM etc.  

The numbering of the GPIOs for PIN variables in ArduPilot is:  

- PWM(1)  50
- PWM(2)  51
- PWM(3)  52
- PWM(4)  53
- PWM(5)  54
- PWM(6)  55
- PWM(7)  56
- PWM(8)  57
- PWM(9)  58
- PWM(10) 59
- PWM(11) 60
- PWM(12) 61
- PWM(13) 62
- PWM(14) 63
- PWM(15) 64
- PWM(16) 65
- CAP1 66
- CAP2 67

## Analog inputs

The JFB-200 has 10 analog inputs

- ADC Pin2  -> +3.3V Sense
- ADC Pin4  -> Battery Voltage 2
- ADC Pin6  -> Battery Current Sensor 2
- ADC Pin8  -> ADC IN 1 (6.6V)
- ADC Pin10 -> RSSI voltage monitoring
- ADC Pin13 -> ADC IN 2 (3.3V)

> - ADC Pin14 -> SERVORAIL sens

Reviewer: sorry, missed this one first pass:

```suggestion
- ADC Pin14 -> SERVORAIL sense
```
